internal/am/upload_transfer.go

```diff
@@ -40,13 +41,15 @@ func (a *UploadTransferActivity) Execute(ctx context.Context, params *UploadTran
 	}
 	defer src.Close()
 
-	bytes, path, err := a.client.Upload(ctx, src, filepath.Base(params.SourcePath))
+	filename := filepath.Base(params.SourcePath)
```

Comment:
I think we should keep the Upload activity unchanged and just take the filepath.Base of the return value of the RemotePath in processing.go.